### PR TITLE
UT: add unit test for ResultSet.join_native

### PR DESCRIPTION
**Pull request by tothegump**
- #5651

### CODE REVIEW

**Comment by tothegump** [comment]
Why another one fails?

**Comment by auvipy** [comment]
> Why another one fails?

the pymongo failing might be the result of new kombu release which switched from pymongo 2.x to 3.x. need to check more in-depth though. do you have the bandwidth to check that?

**Comment by tothegump** [comment]
> > Why another one fails?
> 
> the pymongo failing might be the result of new kombu release which switched from pymongo 2.x to 3.x. need to check more in-depth though. do you have the bandwidth to check that?

Interesting. Seems relates to issue #5676 .
I'd like to have a look.

File: t/unit/tasks/test_result.py [hunk 1]
```diff
@@ -469,6 +469,33 @@ def test_get(self):
         x.get()
         x.join_native.assert_called()
 
+    def test_join_native(self):
+        def _get_meta(tid, result=None, children=None):
+            return {
+                'status': states.SUCCESS,
+                'result': result,
+                # 'traceback': traceback,
```
Comment (thedrow) [inline t/unit/tasks/test_result.py]: No code in comments please.

Comment (tothegump) [reply t/unit/tasks/test_result.py]: Yes, you are right.

**Comment by thedrow** [comment]
@tothegump Can you please explain how this unit test checks that #5496 is fixed?

**Comment by tothegump** [comment]
> @tothegump Can you please explain how this unit test checks that #5496 is fixed?



> It's not clear to me how this test (while it is certainly welcome) is a good test to verify #5496.

In #5496 fails because, when results have GroupResult, which has no "_cache" attr.
Then, join_native can not process correctly this situation which iter_native returns.
So, in this unit test, just mock iter_native returns when GroupResult is an element of results.

**Comment by tothegump** [comment]
I wish I can explain clearly.

**Comment by thedrow** [comment]
Some comments would be nice.
Also, you should rename the test as it doesn't simply test `ResultSet.join_native`.
It does so with very specific conditions.
If you have other test cases in mind, please include them.

**Comment by auvipy** [comment]
――――――――――――――――――――――――― test_GroupResult.test_eq_ne ――――――――――――――――――――――――――

self = <t.unit.tasks.test_result.test_GroupResult object at 0x7f27e6cd9f60>

>   ???

E   AttributeError: 'Celery' object has no attribute 'groupresult'

t/unit/tasks/test_result.py:682: AttributeError

 t/unit/tasks/test_result.py::test_GroupResult.test_eq_ne ⨯

**Comment by tothegump** [comment]
> Some comments would be nice.
> Also, you should rename the test as it doesn't simply test `ResultSet.join_native`.
> It does so with very specific conditions.
> If you have other test cases in mind, please include them.

Make sense. I will do it later.

**Comment by tothegump** [comment]
Looks like, I should do more work for this issue. I mark it as WIP and divide into it:
1. change ut name
2. add comment to this ut;
3. fix the other ut failings.

**Comment by auvipy** [comment]
the UT is passing now, the failing Integration test is different & causing for a kombu issue

**Comment by auvipy** [comment]
this is the root cause of IT failures https://github.com/celery/kombu/commit/052f7601d3a1e09263cc0ebda7ef8c16e6b90175 do you have time to investigate a better fix rather than reverting this?

**Comment by thedrow** [comment]
I'll investigate.

**Pull request merged**